python/aida_interchange/aifutils.py

```diff
@@ -205,83 +209,101 @@ def mark_boundingbox(g, to_mark_on, boundingbox):
 
     :return: The created audio justification resource
     """
-    justification = make_audio_justification(g, doc_id, start_timestamp, end_timestamp, system, confidence)
+    justification = make_audio_justification(
+        g, doc_id, start_timestamp, end_timestamp, system, confidence, uri_ref)
     mark_justification(g, things_to_justify, justification)
 
     return justification
 
 
-def make_keyframe_video_justification(g, doc_id, key_frame, boundingbox, system, confidence):
-    justification = _make_aif_justification(g, doc_id, AIDA_ANNOTATION.KeyFrameVideoJustification, system, confidence)
+def make_keyframe_video_justification(g, doc_id, key_frame, boundingbox, system, confidence, uri_ref=None):
+    justification = _make_aif_justification(
+        g, doc_id, AIDA_ANNOTATION.KeyFrameVideoJustification, system,
+        confidence, uri_ref)
     g.add((justification, AIDA_ANNOTATION.keyFrame,
            Literal(key_frame, datatype=XSD.string)))
     mark_boundingbox(g, justification, boundingbox)
 
     return justification
 
 
-def mark_keyframe_video_justification(g, things_to_justify, doc_id, key_frame, boundingbox, system, confidence):
+def mark_keyframe_video_justification(g, things_to_justify, doc_id, key_frame, boundingbox, system, confidence, uri_ref=None):
     """
     Marks a justification for something appearing in a key frame of a video.
 
     :return: The justification resource
     """
-    justification = make_keyframe_video_justification(g, doc_id, key_frame, boundingbox, system, confidence)
+    justification = make_keyframe_video_justification(g, doc_id, key_frame, boundingbox, system, confidence, uri_ref)
     mark_justification(g, things_to_justify, justification)
 
     return justification
 
 
-def make_shot_video_justification(g, doc_id, shot_id, system, confidence):
-    justification = _make_aif_justification(g, doc_id, AIDA_ANNOTATION.ShotVideoJustification, system, confidence)
+def make_shot_video_justification(g, doc_id, shot_id, system, confidence,
+                                  uri_ref=None):
+    justification = _make_aif_justification(
+        g, doc_id, AIDA_ANNOTATION.ShotVideoJustification, system, confidence,
+        uri_ref)
     g.add((justification, AIDA_ANNOTATION.shot,
            Literal(shot_id, datatype=XSD.string)))
 
     return justification
 
 
-def mark_shot_video_justification(g, things_to_justify, doc_id, shot_id, system, confidence):
+def mark_shot_video_justification(g, things_to_justify, doc_id, shot_id, system,
+                                  confidence, uri_ref):
     """
     Marks a justification for something appearing in a video but not in a key frame.
 
     :return: The justification resource
     """
-    justification = make_shot_video_justification(g, doc_id, shot_id, system, confidence)
+    justification = make_shot_video_justification(g, doc_id, shot_id, system,
+                                                  confidence, uri_ref)
```

Comment:
```suggestion
                                                  confidence, uri_ref=None)
```

Reply:
Got it, I made the change.  I didn't know (or had forgotten) about Examples.py.